components/event-publisher-proxy/pkg/handler/nats/handler.go

```diff
@@ -181,18 +190,26 @@ func (h *Handler) publishCloudEvents(writer http.ResponseWriter, request *http.R
 	// Add tracing context to cloud events
 	tracing.AddTracingContextToCEExtensions(request.Header, event)
 
-	statusCode, dispatchTime, respBody := h.send(ctx, event)
-	h.writeResponse(writer, statusCode, respBody)
+	statusCode, dispatchTime, err := h.send(ctx, event)
+	var errMessage string
+	if err != nil {
+		errMessage = err.Error()
+	}
+	h.writeResponse(writer, statusCode, []byte(errMessage))
 
-	h.namedLogger().With(
+	loggerWithEventInfo := h.namedLogger().With(
 		"id", event.ID(),
 		"source", event.Source(),
 		"before", eventTypeOriginal,
 		"after", eventTypeClean,
 		"statusCode", statusCode,
 		"duration", dispatchTime,
-		"responseBody", string(respBody),
-	).Info("Event dispatched")
+	)
+	if err == nil {
+		loggerWithEventInfo.Info("Event was successfully sent to EventingBackend")
+	} else {
+		loggerWithEventInfo.Error("Failed to send the event to EventingBackend")
+	}
```

Comment:
There is an if statement at line 195 which checking the err, please let's move this if-else logic there.